### DISCUSSION TITLE
Option to Disable Entering free text values into value filter

### DISCUSSION THREAD

**Discussion by jtf9808** [apache/superset#25761] [Ideas]
When using value filters, a user can enter free text into the filter box. 
![Screenshot 2023-10-26 at 10 53 49](https://github.com/apache/superset/assets/70261158/517a6bc7-8847-4fc5-a715-fc201f11d92b)
This is great for quickly selecting pre-existing values in the drop down
![Screenshot 2023-10-26 at 10 54 26](https://github.com/apache/superset/assets/70261158/9efbeb67-dd66-4626-98f4-53a04d3884d5)
 but also allows users to create there own values/categories. 
![Screenshot 2023-10-26 at 10 54 05](https://github.com/apache/superset/assets/70261158/f8ce9b84-3cbd-4fd3-80f9-ee5c589b45f6)
![Screenshot 2023-10-26 at 10 54 15](https://github.com/apache/superset/assets/70261158/51a3d267-2b42-4a30-a0ae-5b5874259d81)

Sometimes this isn't a problem and can be beneficial. However it can often be an issue when we want to limit the categories a user can choose and can also be strange in terms of UX to be able to select invalid values.  

It would be great if there could be an option in the filter settings to restrict selecting/creating custom values (but still be able to use typing to narrow down the existing values, as in the second Image).

**Comment by JurajBurian** [comment]
Hello folks, it is also security issue. 
In case of multi tenant usage (in our case embedded dashboards)  one can try access data that are not mentioned in the list. If lucky, one can see data of other tenants. Think also about sql injection ... in case that one can see queries it is easy.

**Comment by dcincyan** [comment]
We would also like to see this feature added

**Comment by villebro** [comment]
@JurajBurian I don't think this is really a valid security issue - RLS ensures that the the original dataset is filtered, so manually adding a filter that falls outside the values that would otherwise be visible would not give you access to that value. Example:

RLS filter adds the following WHERE clause:

```sql
WHERE colors IN ('red', 'green')
```

If you manually add a filter for `blue`, the WHERE clause would become

```sql
WHERE colors IN ('red', 'green') AND color IN ('blue')
```

This means, that the query would be returning zero rows.

Or am I missing some other detail here? Other than that, adding a checkbox to disable freetext filters IMO makes sense, however, it adds to the already cluttered filter config menu. @kasiazjc @michael-s-molina how do you feel about this?

**Reply by michael-s-molina** [reply]
The main problem with disabling free text is that not all values are loaded in the selector. We currently limit the number of displayed values to a 1000 and users would not be able to select values that are not loaded.

**Reply by JurajBurian** [reply]
Just remark : 
In a new version of Superset (3.1.0)  I am able to see more than 1000 rows in filter -  in my case actually more than 2000.
I use next settings , but not sure what is right : 
FILTER_SELECT_ROW_LIMIT = 10000
NATIVE_FILTER_DEFAULT_ROW_LIMIT = 10000

**Reply by JurajBurian** [reply]
Sometime is disabling free text right solution. 
In original request is a flag suggested, for example freetext = True/False. 
But in case that free text is possible, developer need to understand risk .... SQL Injection is possible then under some circumstances.

**Reply by dcincyan** [reply]
Surely it is possible to allow free text to fetch new filter options dynamically without actually allowing the user to select a free text value

**Reply by JurajBurian** [reply]
@villebro,
rls ensure only fact, that final filter box has only subset of values. 
But if somebody know (or guess) other values, then can "extend" filter with free text (can submit anything) and ask for data that are not allowed for him.
Another level of this issue is potential sql injection.

**Reply by JurajBurian** [reply]
Thanks a lot @dclarke1000, 
I just tested what you proposed and it is working just fine!
-------------------------------------------------------------------------------

Picture shows, more than 1000 options (so one of the settings I mentioned also work as proposed) 
inside and one can type for search but not submit free text. 
![Unsaved Image 3](https://github.com/apache/superset/assets/579632/fbb1d50e-fecc-4c2b-8e54-e6edc8e21423)

p.s. In older version of superset we had some problem with "Dynamically search all filter values " option.
